src/gui/clips/AutomationClipView.cpp

```diff
@@ -100,6 +100,23 @@ void AutomationClipView::changeName()
 
 
 
+void AutomationClipView::connectLastChangedModel()
+{
+	if (AutomatableModel::s_lastChangedModel != nullptr)
+	{
+		bool added = m_clip->addObject(AutomatableModel::s_lastChangedModel);
+		if (!added)
+		{
+			TextFloat::displayMessage(AutomatableModel::s_lastChangedModel->displayName(),
+							tr("Model is already connected to this clip."),
+							embed::getIconPixmap("automation"),
+							2000);
+		}
+		update();
```

Comment:
We call `update` even in the case where the `AutomatableModel` was not added to the `AutomationClip`. Is this expected?

Reply:
I suppose it does not have to be updated if the add was not successful. I only put it there because `AutomationEditor.cpp` had a similar setup, where `update()` is called in `AutomationEditor::dropEvent` regardless of whether the model was added or not. Nevertheless, I have now reworked the if statement to only call `update()` if the add was successful.